### ISSUE TITLE
undefined: serializer.DirectCodecFactory

### ISSUE THREAD

**Issue by withlin** [tektoncd/pipeline#1561]
## go mod
```
k8s.io/api v0.0.0-20191109101513-0171b7c15da1
k8s.io/apimachinery v0.0.0-20191111054156-6eb29fdf75dc
k8s.io/client-go v0.0.0-20191003000419-f68efa97b39e
k8s.io/utils v0.0.0-20191030222137-2b95a09bc58d // indirect
knative.dev/pkg v0.0.0-20191111230022-6335100d292c // indirect
github.com/tektoncd/pipeline v0.8.0
```
## Build Error
```
github.com/tektoncd/pipeline/pkg/client/clientset/versioned/typed/pipeline/v1alpha1
D:\gowork\pkg\mod\github.com\tektoncd\pipeline@v0.8.0\pkg\client\clientset\versioned\typed\pipeline\v1alpha1\pipeline_client.go:104:32: undefined: serializer.DirectCodecFactory
```

ref issue 
[controller-runtim#issues#633](https://github.com/kubernetes-sigs/controller-runtime/issues/633)
[sample-controller#issues#49](https://github.com/kubernetes/sample-controller/issues/49)

>should be upgrade deps?

**Comment by vdemeester** [comment]
@withlin right now pipeline depends on k8s 1.13. We will probably bump those to 1.15 soon but it requires a little bit of work :angel:

**Comment by withlin** [comment]
Thank you for your reply. any progress on go mod in current repo?

**Comment by vdemeester** [comment]
@withlin we are looking into it, that said, it's not gonna happen before kubecon na 2019 for sure, and the main priority is beta apis :angel: *but* there is issues about that to track the work done on it : #1315 and #1538

**Comment by withlin** [comment]
@vdemeester we already  upgrade from  k8s 1.13 to k8s 1.16.2 and then migrate to go mod.

**Comment by wlynch** [comment]
I encountered this when working on #1538, so we can probably close this issue as a duplicate of that. The problem here is that you need to pin all k8s dependencies on the same tag version, which requires a manual override since not all dependencies are following the go mod semver semantics yet.

You can find a WIP prototype here: https://github.com/wlynch/pipeline/tree/modules. The overrides you're interested in can be found here: https://github.com/wlynch/pipeline/blob/modules/go.mod#L86-L102 (though note these are still pinned to k8s v.1.12.9, I still need to bump these to the recent 1.13 upgrade).
